src/ripple/app/ledger/Ledger.cpp

```diff
@@ -100,8 +100,9 @@ class Ledger::sles_iter_impl : public sles_type::iter_base
     bool
     equal(base_type const& impl) const override
     {
-        auto const& other = dynamic_cast<sles_iter_impl const&>(impl);
-        return iter_ == other.iter_;
+        if (auto const p = dynamic_cast<sles_iter_impl const*>(&impl))
+            return iter_ == p->iter_;
+        return false;
```

Comment:
Do we want to check for the case `p == nullptr`?  If we consider this case a logic error then a crash is appropriate (which is what it does with this code).

Reply:
I'm confused? The `if` should never be entered if `p == nullptr` or am I misunderstanding?

```
if (auto const p = dynamic_cast<sles_iter_impl const*>(&impl))
    return iter_ == p->iter_;
return false;
```

Reply:
My bad.  I completely misread it.  Thanks for the correction.